portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -28,11 +35,19 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  private final ArrayList<String> messages =  new ArrayList<>();
-  private final Gson gson = new Gson();
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    List<String> messages =  new ArrayList<>();
+    Gson gson = new Gson();
```

Comment:
Please keep this as final static member variable, and do not initialize everytime.